### PR TITLE
Patch expat

### LINKED ISSUE #1678: Patch expat vulnerabilities
## Description
```
~ ❯ docker run --rm -it --entrypoint "" aquasec/trivy:0.23.0 trivy image aquasec/trivy
2022-02-05T18:42:32.043Z	INFO	Need to update DB
2022-02-05T18:42:32.043Z	INFO	Downloading DB...
21.54 MiB / 21.54 MiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 4.67 MiB p/s 5s
2022-02-05T18:42:43.747Z	INFO	Detected OS: alpine
2022-02-05T18:42:43.747Z	INFO	Detecting Alpine vulnerabilities...
2022-02-05T18:42:43.748Z	INFO	Number of language-specific files: 1
2022-02-05T18:42:43.748Z	INFO	Detecting gobinary vulnerabilities...

aquasec/trivy (alpine 3.15.0)
=============================
Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 2)

+---------+------------------+----------+-------------------+---------------+---------------------------------------+
| LIBRARY | VULNERABILITY ID | SEVERITY | INSTALLED VERSION | FIXED VERSION |                 TITLE                 |
+---------+------------------+----------+-------------------+---------------+---------------------------------------+
| expat   | CVE-2022-23852   | CRITICAL | 2.4.3-r0          | 2.4.4-r0      | expat: integer overflow               |
|         |                  |          |                   |               | in function XML_GetBuffer             |
|         |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-23852 |
+         +------------------+          +                   +               +---------------------------------------+
|         | CVE-2022-23990   |          |                   |               | expat: integer overflow               |
|         |                  |          |                   |               | in the doProlog function              |
|         |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-23990 |
+---------+------------------+----------+-------------------+---------------+---------------------------------------+

usr/local/bin/trivy (gobinary)
==============================
Total: 2 (UNKNOWN: 1, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 1)

+--------------------------------------+------------------+----------+--------------------------------------+---------------+---------------------------------------+
|               LIBRARY                | VULNERABILITY ID | SEVERITY |          INSTALLED VERSION           | FIXED VERSION |                 TITLE                 |
+--------------------------------------+------------------+----------+--------------------------------------+---------------+---------------------------------------+
| github.com/containerd/containerd     | CVE-2021-43816   | CRITICAL | v1.5.8                               | 1.5.9         | containerd: Unprivileged pod          |
|                                      |                  |          |                                      |               | may bind mount any privileged         |
|                                      |                  |          |                                      |               | regular file on disk...               |
|                                      |                  |          |                                      |               | -->avd.aquasec.com/nvd/cve-2021-43816 |
+--------------------------------------+------------------+----------+--------------------------------------+---------------+---------------------------------------+
| github.com/opencontainers/image-spec | GMS-2021-101     | UNKNOWN  | v1.0.2-0.20211117181255-693428a734f5 | 1.0.2         | Clarify `mediaType` handling          |
+--------------------------------------+------------------+----------+--------------------------------------+---------------+---------------------------------------+
```

## What did you expect to happen?
No vulnerabilities

## What happened instead?
2 vulnerabilities were reported

## Output of `trivy -v`:

```
Version: 0.23.0
```

### PR DESCRIPTION
**Pull request by tspearconquest**
Signed-off-by: Thomas Spear <tspear@conquestcyber.com>

## Related issues
- Close #1678

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've followed the [conventions](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by tspearconquest** [comment]
I've submitted #1681 for fixing the Go library vulnerabilities

**Comment by tspearconquest** [comment]
Updated against main after #1681 was merged.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 FROM alpine:3.15.0
-RUN apk --no-cache add ca-certificates git
+RUN apk --no-cache add ca-certificates git expat==2.4.4-r0
```
Comment (AndreyLevchenko) [inline Dockerfile:2]: I ran docker build with default Dockerfile and seems `2.4.4-r0` is already included in alpine:3.15.0
```
Step 1/5 : FROM alpine:3.15.0
 ---> c059bfaa849c
Step 2/5 : RUN apk --no-cache add ca-certificates git
 ---> Running in 3e2990ba22da
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
(1/7) Installing ca-certificates (20211220-r0)
(2/7) Installing brotli-libs (1.0.9-r5)
(3/7) Installing nghttp2-libs (1.46.0-r0)
(4/7) Installing libcurl (7.80.0-r0)
(5/7) Installing expat (2.4.4-r0)
(6/7) Installing pcre2 (10.39-r0)
(7/7) Installing git (2.34.1-r0)
Executing busybox-1.34.1-r3.trigger
Executing ca-certificates-20211220-r0.trigger
```
if it's fixed at os level I would avoid to hardcode dependencies

Comment (tspearconquest) [reply Dockerfile:2]: I agree. This seems to just need a new build of the image to be made in order to patch the vulnerability. I will close this out.

**Pull request closed without merging**